components/resolved_ts/src/endpoint.rs

```diff
@@ -78,55 +80,80 @@ impl ObserveRegion {
         match self.resolver_status {
-            ResolverStatus::Pending { ref mut locks, .. } => {
+            ResolverStatus::Pending {
+                ref mut locks,
+                ref mut tracked_index,
+                ..
+            } => {
                 for log in change_logs {
                     match log {
-                        ChangeLog::Error(e) => return Err(Error::request(e.clone())),
-                        ChangeLog::Rows { rows, .. } => rows.iter().for_each(|row| match row {
-                            ChangeRow::Prewrite { key, start_ts, .. } => {
-                                locks.push(PendingLock::Track {
+                        ChangeLog::Error(e) => {
+                            debug!(
+                                "skip change log error";
+                                "region" => self.meta.id,
+                                "error" => ?e,
+                            );
+                            continue;
+                        }
+                        ChangeLog::Rows { rows, index } => {
+                            rows.iter().for_each(|row| match row {
+                                ChangeRow::Prewrite { key, start_ts, .. } => {
+                                    locks.push(PendingLock::Track {
+                                        key: key.clone(),
+                                        start_ts: *start_ts,
+                                    })
+                                }
+                                ChangeRow::Commit {
+                                    key,
+                                    start_ts,
+                                    commit_ts,
+                                    ..
+                                } => locks.push(PendingLock::Untrack {
                                     key: key.clone(),
                                     start_ts: *start_ts,
-                                })
-                            }
-                            ChangeRow::Commit {
-                                key,
-                                start_ts,
-                                commit_ts,
-                                ..
-                            } => locks.push(PendingLock::Untrack {
-                                key: key.clone(),
-                                start_ts: *start_ts,
-                                commit_ts: *commit_ts,
-                            }),
-                            // One pc command do not contains any lock, so just skip it
-                            ChangeRow::OnePc { .. } => {}
-                        }),
+                                    commit_ts: *commit_ts,
+                                }),
+                                // One pc command do not contains any lock, so just skip it
+                                ChangeRow::OnePc { .. } => {}
+                            });
+                            assert!(*tracked_index < *index);
```

Comment:
Probably should show more information like region id and their actual index value.